pom.xml

```diff
@@ -270,9 +270,8 @@
     <dependencies>
       <dependency>
         <groupId>org.apache.spark</groupId>
-        <artifactId>spark-test-tags_${scala.binary.version}</artifactId>
+        <artifactId>spark-tags_${scala.binary.version}</artifactId>
         <version>${project.version}</version>
-        <scope>test</scope>
```

Comment:
The key change here is that now, most usages of this modules are usages of _its test code_, _from other test code_. This declaration is right, but would only be used by `core` and anything else that doesn't depend on `core`. They're the only ones with a dependency on its _non test code_ (`@Since`) from their _non test code_.

I think the parent also has to make a separate declaration about this same module but with `<classifier>tests</classifier>` This configures how things depends on its test code. Then, modules that use the test-related code here need a dependency with `<scope>test</scope>` and `<classifier>tests</classifier>`.